src/libs/decoders/mp3_seek_table.cpp

```diff
@@ -19,188 +19,22 @@
-    const auto rc = stat(filename, &stat_buf);
-    return (rc >= 0) ? stat_buf.st_size : -1;
-}
-
-
-// Calculates a unique 64-bit hash (integer) from the provided file.
-// This function should not cause side-effects; ie, the current
-// read-position within the file should not be altered.
-//
-// This function tries to files as-close to the middle of the MP3 file as possible,
-// and use that feed the hash function in hopes of the most uniqueness.
-// We're trying to avoid content that might be duplicated across MP3s, like:
-// 1. ID3 tag filler content, which might be boiler plate or all empty
-// 2. Trailing silence or similar zero-PCM content
-//
-uint64_t calculate_stream_hash(struct SDL_RWops* const context) {
-    // Save the current stream position, so we can restore it at the end of the function.
-    const auto original_pos = SDL_RWtell(context);
-
-    // Seek to the end of the file so we can calculate the stream size.
-    SDL_RWseek(context, 0, RW_SEEK_END);
-    const auto end_pos = SDL_RWtell(context);
-    SDL_RWseek(context, original_pos, RW_SEEK_SET); // restore position
-
-    // Check if we can trust the end position as the stream size
-    if (end_pos <= 0)
-        return 0;
-    const auto stream_size = static_cast<size_t>(end_pos);
-
-    // create a hash state
-    const auto state = XXH64_createState();
-    if (!state)
-        return 0;
-
-    // Initialize state seeded with our total stream size
-    const auto seed = static_cast<XXH64_hash_t>(stream_size);
-    if (XXH64_reset(state, seed) == XXH_ERROR) {
-        XXH64_freeState(state);
-        return 0;
-    }
-
-    // Setup counters and read buffer
-    XXH64_hash_t hash = 0;
-    size_t current_bytes_read = 0;
-    size_t total_bytes_read = 0;
-    auto hash_state = XXH_OK;
-    vector<char> buffer(1024, 0);
-
-    // Seek prior to the last 32 KB (or less) in the file, which is what we'll hash
-    const auto bytes_to_hash = std::min(static_cast<size_t>(32768u), stream_size);
-    const auto pos_to_hash_from = static_cast<Sint64>(stream_size - bytes_to_hash);
-    SDL_RWseek(context, pos_to_hash_from, RW_SEEK_SET);
-    assert(SDL_RWtell(context) == pos_to_hash_from);
-
-    // Feed the state with input data, any size, any number of times
-    do {
-        current_bytes_read = SDL_RWread(context, buffer.data(), 1, buffer.size());
-        total_bytes_read += current_bytes_read;
-        hash_state = XXH64_update(state, buffer.data(), current_bytes_read);
-    }
-    while (current_bytes_read
-           && total_bytes_read < bytes_to_hash
-           && hash_state == XXH_OK);
-
-    // Finalize the hash and clean it up
-    hash = XXH64_digest(state);
-    XXH64_freeState(state);
-
-    // restore the stream position and cleanup the hash
-    SDL_RWseek(context, original_pos, RW_SEEK_SET);
-    assert(SDL_RWtell(context) == original_pos);
-
-    return hash;
-}
-
 // This function generates a new seek-table for a given mp3 stream and writes
```

Comment:
Could you update the date in the header comment to 2025?

Reply:
Updated